test/test.baseexternalclient.ts

```diff
@@ -82,6 +82,24 @@ describe('BaseExternalAccountClient', () => {
     client_id: 'CLIENT_ID',
     client_secret: 'SECRET',
   };
+  const externalAccountOptionsWorkforceUserProject = Object.assign(
```

Comment:
I think we should test the following scenarios:
- Workforce audience + workforce_pool_user_project + no impersonation
- Workforce audience + workforce_pool_user_project + client auth + no impersonation => user project not passed
- Workforce audience + client auth + no impersonation => confirm no error thrown
- Workforce audience + workforce_pool_user_project + impersonation => confirm before impersonation userProject passed

It is important we test with no impersonation. Workforce pools are most commonly used without impersonation.